protostar/commands/test/cheatcodes/declare_cheatcode.py

```diff
@@ -0,0 +1,60 @@
+import asyncio
+from pathlib import Path
+from typing import cast
+from dataclasses import dataclass
+
+
+from starkware.python.utils import from_bytes
+from starkware.starknet.testing.contract import DeclaredClass
+from starkware.starknet.testing.contract_utils import get_abi, get_contract_class
+
+
+from starkware.starknet.business_logic.internal_transaction import (
+    InternalDeclare,
+)
+
+from protostar.commands.test.cheatcodes.cheatcode import Cheatcode
+
+
+@dataclass
+class DeclaredContract:
+    class_hash: int
+
+
+class DeclareCheatcode(Cheatcode):
+    @staticmethod
+    def name() -> str:
+        return "declare"
+
+    @staticmethod
+    def implementation() -> str:
+        return "declare"
```

Comment:
I don't like this approach because:
- `implementation` is not the best name
- it's not type safe
- makes refactoring harder
- potential errors will be catched by unit tests rather than pyright what increases feedback loop

I'd rather use `build` or `execute` approach. 
`build` method can return `Callable[[Any],Any]`